### PR TITLE
Remove unnecessary code

### PR DESCRIPTION
**Pull request by swissspidy**
### Description

Reduces code complexity.

### Type of change
Please select the relevant options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Steps to test

Verify that tests still pass.

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [x] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by felixarntz — APPROVED** [review]
Nice! +111111111

This was also striking my eye seeming unnecessary, but wasn't sure it could easily be dropped.

File: inc/providers/class-wp-sitemaps-posts.php [hunk 1]
```diff
@@ -130,7 +126,7 @@ public function get_url_list( $page_num, $post_type = '' ) {
 	 */
 	public function max_num_pages( $post_type = '' ) {
 		if ( empty( $post_type ) ) {
-			$post_type = $this->get_queried_type();
+			return 1;
```
Comment (felixarntz) [inline inc/providers/class-wp-sitemaps-posts.php]: Should this better be 0?

File: inc/providers/class-wp-sitemaps-taxonomies.php [hunk 2]
```diff
@@ -126,7 +116,7 @@ public function get_url_list( $page_num, $taxonomy = '' ) {
 	 */
 	public function max_num_pages( $taxonomy = '' ) {
 		if ( empty( $taxonomy ) ) {
-			$taxonomy = $this->get_queried_type();
+			return 1;
```
Comment (felixarntz) [inline inc/providers/class-wp-sitemaps-taxonomies.php]: Same here, maybe better as 0?

File: inc/providers/class-wp-sitemaps-posts.php (hunk 1 above)
Comment (swissspidy) [reply inc/providers/class-wp-sitemaps-posts.php]: makes sense. fixed!

**Pull request merged**